alerts/cluster-monitoring-operator/TelemeterClientFailures.md

```diff
@@ -0,0 +1,99 @@
+# TelemeterClientFailures
+
+## Meaning
+
+The alert `TelemeterClientFailures` is triggered when the Telemeter client fails
+to send Telemetry data at a certain rate over a period of time
+to Red Hat.
+
+The `telemeter-client` pod running in the `openshift-monitoring`
+namespace collects [selected platform metrics](https://docs.openshift.com/container-platform/latest/support/remote_health_monitoring/showing-data-collected-by-remote-health-monitoring.html#showing-data-collected-from-the-cluster_showing-data-collected-by-remote-health-monitoring)
+from the `prometheus-k8s` service at
+regular intervals using the `/federate` endpoint and ships them periodically
+to Red Hat using a custom secured protocol.
+
+## Impact
+
+When the alert fires, Red Hat support and engineering teams don't have a complete
+view of the cluster anymore. It may hinder the ability for Red Hat to
+proactively detect issues in the cluster.
+
+## Diagnosis
+
+* Review the logs for the pod `telemeter-client`
+in the `openshift-monitoring` namespace.
+
+You can review the logs for the `telemeter-client` pod in the
+`openshift-monitoring` namespace by running the following command:
+
+```console
+$ oc logs -n openshift-monitoring deployment.apps/telemeter-client -c telemeter-client | less
+```
+
+* Execute the PromQL expressions to identify which side issue is happening
+
+  * 4.17+
+
+    ```console
+    sum by(client, status_code) (rate(metricsclient_http_requests_total{status_code!~"2.."})) > 0
+    ```
+
+    * `client` label value shows which side is faulty.
+    * `federate_from` when the error is from Prometheus.
+    * `federate_to` when the error is from Telemeter server.
+
+  * 4.16 and below
+
+    Check Prometheus status
+
+      ```console
+      sum by(client, status_code) (rate(metricsclient_request_retrieve{status_code!~"200"}[5m])) > 0
+      ```
+
+    Check Telemeter Server status
+
+      ```console
+      sum by(client, status_code) (rate(metricsclient_request_send{status_code!~"200"}[5m])) > 0
+      ```
+
+## Mitigation
+
+The resolution depends on the particular issue reported in the logs.
+
+* Telemeter client fails to pull metrics from Prometheus.
+  * You'd need to check the availability of Prometheus.
+
+* Telemeter client fails to push to the server.
+
+  * Make sure [HTTP proxies allow](https://docs.openshift.com/container-platform/latest/installing/install_config/configuring-firewall.html)
+  the telemeter server names.
+
+  * It can be due to Red Hat Telemeter server outage.
```

Comment:
Should we ask users to check the `status_code` label value as it would be more convenient that looking through the logs?
Another follow-up improvement could be to have new alerting rules based on the `metricsclient_http_requests_total` metric which would tell the difference between 4xx errors (e.g. bad pull secret) and 5xx (issue on Red Hat side).

Reply:
ya I can create a separate jira for that as this one is for runbook

Reply:
https://github.com/prometheus-operator/prometheus-operator/issues/6948